sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1028,15 +1028,19 @@ _pi_queue::resetCommandList(pi_command_list_ptr_t CommandList,
     EventList.clear();
   } else {
     // For immediate commandlist reset only those events that have signalled.
+    std::vector<pi_event> UnsetEventsList{};
     for (auto it = EventList.begin(); it != EventList.end(); it++) {
       std::scoped_lock<pi_shared_mutex> EventLock((*it)->Mutex);
       ze_result_t ZeResult =
           ZE_CALL_NOCHECK(zeEventQueryStatus, ((*it)->ZeEvent));
       if (ZeResult == ZE_RESULT_SUCCESS) {
-        std::move(it, it, std::back_inserter(EventListToCleanup));
-        EventList.erase(it, it);
+        EventListToCleanup.push_back(std::move((*it)));
+      } else {
+        UnsetEventsList.push_back(std::move((*it)));
       }
     }
+    EventList.clear();
+    EventList = std::move(UnsetEventsList);
```

Comment:
please explain what this is solving. why is it not in a separate PR?

Reply:
This fixes an event leak. I can put it in a different PR if you like.

Reply:
yes, please

Reply:
OK